### ISSUE TITLE
Provider produced inconsistent final plan when expanding the plan for azurerm_network_interface_application_gateway_backend_address_pool_association

### ISSUE THREAD

**Issue by arestarh** [terraform-providers/terraform-provider-azurerm#5928]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

**terraform** 0.12.20
**AzureRM** provider version 1.44.0
### Affected Resource(s)

* `azurerm_network_interface_application_gateway_backend_address_pool_association`

### Terraform Configuration Files

```hcl
resource "azurerm_network_interface_application_gateway_backend_address_pool_association" "example_association" {
  network_interface_id    = azurerm_network_interface.example_nic.id
  ip_configuration_name   = "example-ipconf"
  backend_address_pool_id = azurerm_application_gateway.agw00.backend_address_pool[1].id
}
```

### Panic Output

> Error: Provider produced inconsistent final plan 
> When expanding the plan for azurerm_network_interface_application_gateway_backend_address_pool_association.example_association 
> to include new values learned so far during apply, provider "registry.terraform.io/-/azurerm" produced an invalid new value for 
> .backend_address_pool_id: was cty.StringVal("/subscriptions/********/resourceGroups/exp-rg/providers/Microsoft.Network/applicationGateways/agw00/backendAddressPools/pool1"), 
> but now cty.StringVal("/subscriptions/********/resourceGroups/exp-rg/providers/Microsoft.Network/applicationGateways/agw00/backendAddressPools/pool2"). 
> This is a bug in the provider, which should be reported in the provider's own issue tracker. 

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by tombuildsstuff** [comment]
hi @arestarh 

Thanks for opening this issue.

Azure's changed the behaviour of the Network Interface <-> Application Gateway association so that rather than being per IP Configuration this is now per Network Interface.

Whilst it's unfortunate this is a breaking behavioural change in the Azure API, version 2.0 of the Azure Provider updates to account for this by dropping the `ip_configuration_name` field from this resource - and as such this should be resolved by updating to version 2.0 of the Azure Provider.

Since this should be fixed by moving to version 2.0 of the Azure Provider I'm going to close this issue for the moment but please let us know if that doesn't work for you and we'll take another look.

Thanks!

**Comment by arestarh** [comment]
> hi @arestarh
> 
> Thanks for opening this issue.
> 
> Azure's changed the behaviour of the Network Interface <-> Application Gateway association so that rather than being per IP Configuration this is now per Network Interface.
> 
> Whilst it's unfortunate this is a breaking behavioural change in the Azure API, version 2.0 of the Azure Provider updates to account for this by dropping the `ip_configuration_name` field from this resource - and as such this should be resolved by updating to version 2.0 of the Azure Provider.
> 
> Since this should be fixed by moving to version 2.0 of the Azure Provider I'm going to close this issue for the moment but please let us know if that doesn't work for you and we'll take another look.
> 
> Thanks!

Thanks a lot for explaining this case.
We are going to migrate to v2.0 if provider soon. And if we experience similar issue I'll get back to this gurhub issue.
